docs/docs/building-ui/composite-components.md

```diff
@@ -1,58 +1,146 @@
 
 ```java
 public class ApplicationComponent extends Composite<Div> {
 	//Implementation
 }
 ```
 
-## Component binding
+## Constructor setup (Recommended)
+
+The recommended approach is to configure everything in the constructor. Use `getBoundComponent()` to build your internal layout and logic directly.
+
+<ComponentDemo 
+path='/webforj/composite?' 
+cssURL='https://raw.githubusercontent.com/webforj/webforj-documentation/main/src/main/resources/static/composite/composite.css'
+javaE='https://raw.githubusercontent.com/webforj/webforj-documentation/refs/heads/main/src/main/java/com/webforj/samples/views/composite/CompositeView.java'
+height='400px'
+/>
 
-The `Composite` class requires developers to specify the type of `Component` it manages. This strong association ensures that a `Composite` component is intrinsically linked to its underlying Component. This also provides benefits over traditional inheritance, as it allows the developer to decide exactly what functionality to expose to the public API. 
+`getBoundComponent()` works perfectly in the constructor - no lifecycle methods needed for most use cases.
 
-By default, the `Composite` component utilizes the generic type parameter of its subclass to identify and instantiate the bound component type. This is based on the assumption that the component class has a parameter-less constructor. Developers can customize the component initialization process by overriding the `initBoundComponent()` method. This allows for greater flexibility in creating and managing the bound component, including invoking parameterized constructors.
+## Component binding
+
+The `Composite` class enforces a strong link between the custom component and its underlying component. By default, it instantiates the bound component using its no-argument constructor. If you need more control, you can override this behavior.
 
-The following snippet overrides the initBoundComponent method to use a parameterized constructor for the [FlexLayout](../components/flex-layout.md) class:
+To create the bound component manually with custom parameters or default children, override the `initBoundComponent()` method.
 
 ```java
 public class OverrideComposite extends Composite<FlexLayout> {
-	
-	TextField nameField;
-	Button submit;
+    TextField nameField;
+    Button submit;
 
-	@Override
-	protected FlexLayout initBoundComponent() {
-		nameField = new TextField();
-		submit = new Button("Submit");
-		return new FlexLayout(nameField, submit);
-	}
+    @Override
+    protected FlexLayout initBoundComponent() {
+        nameField = new TextField();
+        submit = new Button("Submit");
+        return new FlexLayout(nameField, submit);
+    }
 }
 ```
 
-## Lifecycle management
+This override lets you pass parameters to the constructor, add default children, or set up custom layout behavior.
```

Comment:
Check the Javadocs. Composite uses the generic type of its subclass to figure out and create the bound component, as long as that component has a no-argument constructor. If needed, subclasses can override the initBoundComponent() method to customize how the component is created.

But you usually only need to do this if the component you're binding has a custom generic type. It's not meant for customizing the bound component in terms of adding children or changing its structure.